easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -323,7 +325,7 @@ def __init__(self, path, extra_options=None, build_specs=None, validate=True, hi
 
         # list of *all* dependencies, including hidden/build deps & toolchain, but excluding filtered deps
         self.all_dependencies = copy.deepcopy(self.dependencies())
-        if self.toolchain.name != DUMMY_TOOLCHAIN_NAME:
+        if self['toolchain']['name'] != DUMMY_TOOLCHAIN_NAME:
```

Comment:
no, it's avoiding that a `Toolchain` instance is created already at this point, which is not needed at all for `dummy` toolchains here...